Caltech-IPAC/firefly
# DM-16036: Persist added information in DataGroup to database then send them to client

https://jira.lsstcorp.org/browse/DM-16036

- Added PARAMs and PARAMrefs to GroupInfo from VoTable.
- Preserve the additional meta during its data flow: VoTable -> DataGroup -> database -> DataGroup -> JSON string -> client
- Added VotableTest for unit testing
- update jsdocs to include the new meta info, `tables-typedefs.jsdoc`

Cleanup:
- remove TableDef from main SearchProcessor flow.  It is now renamed to IpacTableDef, since it's only used for IPAC table.
- delete a couple of files related to ResultSet -> DataGroup translation.  Added a simple, generic conversion routine instead.

NOTE:  Some of the meta did make it to the client when using catalog upload or via 'Upload' menu due to ticket DM-16273.
 
Testing: 
https://irsawebdev9.ipac.caltech.edu/dm-16036/firefly/
- Please help with regression test.  The cleanup works that may break existing code.
- As far as the new meta info, there's no UI added.  It's best to look at the test to see if it adequately cover the requirement.
